### PR TITLE
Breaking: Feature/appID

### PR DESCRIPTION
**Pull request by Shahar-Y**
Adds the AppID. needs to be pushed with: 
 * api-gw: feature/oauth : https://github.com/meateam/api-gateway/pull/143
 * web-ui: feature/appID

### CODE REVIEW

File: src/file/file.service.spec.ts [hunk 1]
```diff
@@ -1,933 +1,933 @@
-import * as chai from 'chai';
```
Comment (Shahar-Y) [inline src/file/file.service.spec.ts:1]: WHY WOULD YOU DO THAT????????

File: src/file/file.service.ts [hunk 2]
```diff
@@ -37,6 +37,7 @@ export class FileService {
     type: string,
     folderID: string = '',
     key: string | null = null,
+    appID: string,
```
Comment (Shahar-Y) [inline src/file/file.service.ts:40]: put appID before folderID. defaults come last

File: src/file/file.service.spec.ts (hunk 1 above)
Comment (nogahaskal) [reply src/file/file.service.spec.ts]: done.

File: src/file/file.service.ts (hunk 2 above)
Comment (nogahaskal) [reply src/file/file.service.ts]: done commander

**Comment by Shahar-Y** [comment]
Resolve the conflicts and it's approved @nogahaskal

File: src/utils/errors/client.error.ts [hunk 3]
```diff
@@ -43,6 +43,12 @@ export class FileExistsWithSameName extends ClientError {
   }
 }
 
+export class FileParentAppIDNotEqual extends ClientError {
+  constructor(message?: string) {
+    super(message || 'file parent app id is not the same', grpc.status.INVALID_ARGUMENT);
```
Comment (AllKinds) [inline src/utils/errors/client.error.ts:48]: The message is not clear enough. I think you should also add the different parent and file appIDs.

Comment (AllKinds) [reply src/utils/errors/client.error.ts:48]: Also, I'm not sure that `INVALID_ARGUMENT` is the appropriate error type. Maybe it should be `PERMISSION_DENIED`.

File: src/file/file.model.ts [hunk 4]
```diff
@@ -121,14 +125,20 @@ fileSchema.pre('save', async function (next: NextFunction) {
...
+  const query: any = { name, parent, appID };
   if (!parent) {
     query.ownerID = ownerID;
   }
 
   const existingFile = await fileModel.findOne(query);
-  if (existingFile && !existingFile.float) {
+
+  if (parentFile && parentFile.appID !== appID) {
+    next(new FileParentAppIDNotEqual());
```
Comment (AllKinds) [inline src/file/file.model.ts:140]: Note that this means that the main drive-client wont be able to create files inside other applications folders.

Comment (Shahar-Y) [reply src/file/file.model.ts:140]: That's OK for now

File: src/utils/errors/client.error.ts (hunk 3 above)
Comment (nogahaskal) [reply src/utils/errors/client.error.ts]: done.

**Pull request merged**